core/templates/dev/head/pages/exploration_player/PlayerTranscriptService.js

```diff
@@ -20,146 +20,146 @@
 // are shown, what answers have been given, etc. Note that this service does
 // not maintain the currently-active card -- it's more like a log of what the
 // learner has 'discovered' so far.
-oppia.factory('PlayerTranscriptService', ['$log', function($log) {
-  // Each element of this array represents a 'card' in the learner view,
-  // represented as a JavaScript object with the following keys:
-  // - stateName: the name of the state
-  // - currentParams: an object with the current parameter names and values.
-  //   Each object has two keys:
-  //   - parameterName: the name of the parameter
-  //   - parameterValue: the new value of the parameter
-  // - contentHtml: the HTML representing the non-interactive content, i.e.
-  //     what Oppia first says to the learner before asking for a response
-  // - interactionHtml: the HTML representing the interaction
-  // - inputResponsePairs: a list of input response pairs:
-  //   - learnerInput: the JS representation of the learner's input. This can
-  //       either be an answer or a request for a hint.
-  //   - oppiaResponse: the HTML representation of Oppia's response to
-  //       the learner's input. This could either be a hint or a feedback for
-  //       the learner's answer.
-  //   - isHint: A boolean value representing if the current input is a request
-  //       for a hint.
-  // - destStateName: if non-null, this means that the learner is ready to move
-  //     on. It represents the state name of the next card.
-  //
-  // Note that every card in this transcript is visible on the screen. The
-  // 'destStateName' field is intended to identify transcripts where there is a
-  // card 'in reserve', but the learner has not yet navigated to it -- this
-  // happens if the current card offers feedback to the learner before they
-  // carry on.
-  var transcript = [];
-  var numAnswersSubmitted = 0;
+oppia.factory('PlayerTranscriptService', [
+  '$log', 'StateCardObjectFactory', function($log, StateCardObjectFactory) {
+    // Each element of this array represents a 'card' in the learner view,
```

Comment:
Hm, shouldn't it now be an actual StateCard object? If so you can remove all the detailed internal description.

Reply:
Done.